Hazelcast.Net/Hazelcast.Client.Connection/ClientConnectionManager.cs

```diff
@@ -311,10 +311,6 @@ private void Authenticate(ClientConnection connection, bool isOwnerConnection)
             }
             var result = ClientAuthenticationCodec.DecodeResponse(response);
 
-            if (result.address == null)
```

Comment:
Why are we removing this ? Address.decode can return null anytime. On java if we get null here we are handling it as failure. Otherwise it is resulting in an unstable intermediate state. I think, We should keep the check and the exception.